nifi-bootstrap/src/test/java/org/apache/nifi/bootstrap/util/TestSecureNiFiConfigUtil.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.bootstrap.util;
+
+import org.apache.nifi.properties.NiFiPropertiesLoader;
+import org.apache.nifi.security.util.KeyStoreUtils;
+import org.apache.nifi.security.util.TlsException;
+import org.apache.nifi.util.NiFiProperties;
+import org.bouncycastle.asn1.x509.GeneralName;
+import org.bouncycastle.util.IPAddress;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.bind.DatatypeConverter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.SignatureException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class TestSecureNiFiConfigUtil {
+    private Logger logger = LoggerFactory.getLogger("org.apache.nifi.bootstrap.util.TestSecureNiFiConfigUtil");
+
+    private static final boolean EXPECT_STORES_TO_EXIST = true;
+
+    private String nifiPropertiesFile = "src/test/resources/nifi.properties";
```

Comment:
Recommend evaluating Class.getResource() as an alternative the specifying the full `src/test/resources` path here and in other places.

Reply:
I'll do this with nifi.properties, but the keystore and truststore paths are referenced from within the nifi.properties file, and since the SecureNiFiConfigUtil uses file paths instead of Class.getResource(), I think it makes more sense to keep the reference to src/test/resources in the keystore/truststore cases.  Let me know if you see any other options for those cases.

Reply:
Interestingly, though, in the nifi.properties case, Class.getResource() won't work, because it's a temporary file that I create on the fly during the test, and there of course isn't a resource yet to get when I want to get the Path to that file using the Class.getResource() method.  So I'm going to suggest we keep it as is, and I'll just extract src/test/resources/ into a constant.  How does that sound?

Reply:
The file in `src/test/resources` will need to be copied to be a temporary location and updated with the generated paths, so Class.getResourceAsStream() could be used to read the initial file.  Alternatively, if it is simple enough to just write the minimum properties to a temporary file, perhaps the `nifi.properties` in `src/test/resources` may not be necessary at all.